plugins/by-name/harpoon/deprecations.nix

```diff
@@ -0,0 +1,72 @@
+{ lib, ... }:
+{
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "harpoon"
+      ];
+
+      commonWarning = ''
+        /!\ `plugins.harpoon` has been refactored to now use harpoon2 (https://github.com/ThePrimeagen/harpoon/tree/harpoon2).
+      '';
+
+      optionNames = [
+        "saveOnToggle"
+        "saveOnChange"
+        "enterOnSendcmd"
+        "tmuxAutocloseWindows"
+        "excludedFiletypes"
+        "markBranch"
+        "projects"
+        "menu"
+      ];
+    in
+    (map (
+      optionName:
+      lib.mkRemovedOptionModule (basePluginPath ++ [ optionName ]) ''
+        ${commonWarning}
+
+        You may now use `plugins.harpoon.settings` option to forward any value to the `require("harpoon"):setup()` call.
+      ''
+    ) optionNames)
+    ++ [
+      (lib.mkRemovedOptionModule (basePluginPath ++ [ "keymaps" ]) ''
+        ${commonWarning}
+
+        The `plugins.harpoon` module no longer allows you to define your keymaps.
+        Please, manually define your keymaps using the top-level `keymaps` option.
+
+        For example,
+        ```
+          plugins.harpoon.keymaps = {
+            addFile = "<leader>a";
+            toggleQuickMenu = "<C-e>";
+            navFile = {
+              "1" = "<C-j>";
+              "2" = "<C-k>";
+              "3" = "<C-l>";
+              "4" = "<C-m>";
+            };
+          };
+        ```
+
+        would become:
+        ```
+          keymaps = [
+            { mode = "n"; key = "<leader>a"; action.__raw = "harpoon:list():add"; }
```

Comment:
Is this accurate? Wouldn't it need `require('harpoon')` given I assume we don't have a `harpoon` variable in scope?